chunk/schema.go

```diff
@@ -106,8 +106,8 @@ func (cfg SchemaConfig) hourlyBuckets(from, through model.Time, userID string, m
 	)
 
 	for i := fromHour; i <= throughHour; i++ {
-		relativeFrom := util.Max64(i*millisecondsInHour, int64(from))
-		relativeThrough := util.Min64((i+1)*millisecondsInHour, int64(through))
+		relativeFrom := util.Max64(0, int64(from)-(i*millisecondsInHour))
```

Comment:
I'm having trouble understanding how this works. First, the `relative...` naming is confusing - if I understand correctly, the resulting `relativeFrom` and `relativeThrough` timestamps are used as absolute bucket timestamps in the schema's `GetWriteEntries()` etc. functions. So if that's true, how can it be right to start with a `from` that's X hours earlier than the original `from` and then just go earlier and earlier from there, instead of later? Same for `through`. Also, shouldn't bucket timestamps be hourly-aligned? I don't see how that can be the case when the unaligned `from` is simply added as a term to an aligned value here.

Gah, I'm probably just missing something stupid, but my head doesn't want to wrap around it today?